### ISSUE TITLE
ignore pricing rule not used in selling workflow

### ISSUE THREAD

**Issue by melBoh** [frappe/erpnext#30986]
### Information about bug

Hi, 
I've got a serious issue with item prices. 
How to reproduce :
- Create a quotation, check Ignore pricing rule 
- Edit the items prices to make them custom
- save, submit and make a sales order from quotation

Result : Ignore pricing rule is not checked and custom prices have not followed.
Same thing happens if you create an invoice from an order with custom prices.
I think it comes from [this commit ](https://github.com/frappe/erpnext/commit/1f79b47a1797762cd57972f6f3b28688d558a0cc)
it seems some experienced the same issue because i found [this revert](https://github.com/frappe/erpnext/pull/30363)
but it's still present in my current version
What should i do ? will the revert be merged in a future release ? should i cherry-pick it in the meanwhile ? Or maybe revert to a previous version ?
Thank you !

### Module

selling

### Version

ERPNext: v13.29.0 (version-13)
Frappe Framework: v13.29.0 (version-13)

**Comment by melBoh** [comment]
Hi ! 
I still have this issue :(
The answer I had on https://github.com/frappe/erpnext/pull/30363 sadly isn't working for me
Here you can see a simple sales order with 1 item which has a pricing rule that I want to ignore
![commande-6articles](https://user-images.githubusercontent.com/15611886/171658518-a2c49902-d666-46a8-99df-4ac419b82a73.png)
And here is the invoice generated
![Capture d’écran 2022-06-02 à 16 06 33](https://user-images.githubusercontent.com/15611886/171658928-aa4be701-ea65-4e00-8bff-478ca6a11a59.png)
Ignore pricing rule is NOT checked, thus pricing rules are applied, and you can probably imagine the kind of accounting mayhem it makes.
Yes I can check  Ignore pricing rule myself (however it's bound to be forgotten once in a while), but it's even more problematic when I use custom prices on the order. In this case it's wiped up when I create the invoice and I need to rewrite them manually...
Please can anybody try it on their ERPNext v13 up to date ? 
Thank you

**Comment by nidhipurohit11** [comment]
When I checked, the custom prices are followed, but yeah the checkbox "Ignore Pricing Rule" remains unchecked on SO.

**Comment by melBoh** [comment]
Thank you @nidhipurohit11 ! 
Here prices follow as long as I don't have pricing rules available for them. If there is, the pricing rule IS applied (probably because the checkbox is not checked)
For now I override the "make" functions that I need, but if it is replicable, then maybe it can be tagged as a bug ?

For those who are interested, here is my hack
in hooks.py
```
override_whitelisted_methods = {
	"erpnext.selling.doctype.sales_order.sales_order.make_sales_invoice": "vendor.module.override.sales_order.make_sales_invoice",
	[...]
}
```
then in <your vendor>/<your module>/override/sales_order.py

```
@frappe.whitelist()
def make_sales_invoice(source_name, target_doc=None, ignore_permissions=False):
	from erpnext.selling.doctype.sales_order.sales_order import make_sales_invoice
	if not target_doc:
		target_doc = frappe.new_doc("Sales Invoice")
	target_doc.ignore_pricing_rule = 1
	return make_sales_invoice(source_name, target_doc)

```

I did this for every document concerned by [this commit](https://github.com/frappe/erpnext/commit/1f79b47a1797762cd57972f6f3b28688d558a0cc) that I use

**Comment by maharshivpatel** [comment]
@melBoh this issue is fixed on latest version of ERPNext (13.39.2) . please update and let me know if problem still persists.
